src/OpenTelemetry/Trace/Configuration/OpenTelemetryBuilder.cs

```diff
@@ -0,0 +1,71 @@
+﻿// <copyright file="OpenTelemetryBuilder.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Trace.Configuration
+{
+    /// <summary>
+    /// Build OpenTelemetry pipeline.
+    /// Currently supports a single processing pipeline, and any number of activity source names
+    /// to subscribe to.
+    /// </summary>
+    public class OpenTelemetryBuilder
+    {
+        internal OpenTelemetryBuilder()
+        {
+        }
+
+        internal ActivityProcessorPipelineBuilder ProcessingPipeline { get; private set; }
+
+        internal HashSet<string> ActivitySourceNames { get; private set; }
+
+        /// <summary>
+        /// Sets processing and exporting pipeline.
+        /// </summary>
+        /// <param name="configure">Function that configures pipeline.</param>
+        /// <returns>Returns <see cref="OpenTelemetryBuilder"/> for chaining.</returns>
+        public OpenTelemetryBuilder SetProcessorPipeline(Action<ActivityProcessorPipelineBuilder> configure)
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            var pipelineBuilder = new ActivityProcessorPipelineBuilder();
+            configure(pipelineBuilder);
+            this.ProcessingPipeline = pipelineBuilder;
+            return this;
+        }
+
+        /// <summary>
+        /// Adds given activitysource name to the list of subscribed sources.
+        /// </summary>
+        /// <param name="activitySourceName">Activity source name.</param>
+        /// <returns>Returns <see cref="OpenTelemetryBuilder"/> for chaining.</returns>
+        public OpenTelemetryBuilder AddActivitySource(string activitySourceName)
+        {
+            if (this.ActivitySourceNames == null)
+            {
+                this.ActivitySourceNames = new HashSet<string>();
+            }
+
+            this.ActivitySourceNames.Add(activitySourceName.ToLowerInvariant());
```

Comment:
My understanding is that it's always better to use ToUpperInvariant? [Turkey Test](http://www.moserware.com/2008/02/does-your-code-pass-turkey-test.html), basically.

Reply:
Thanks! 
We could install and enable some codeanalysis tools to catch this. If one exists.

Reply:
FYI there's another one here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/660/files#diff-6a33a0caf6b7f19c4dd84455e70ac2c8R66

Reply:
I think there is a CA warning for this, by the way. https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1308?view=vs-2019 maybe?

Reply:
Yes. I meant some automatically checks as part of CI to catch this.

Reply:
I changed this code, so closing this conversation as resolved.